app/src/main/java/com/asfoundation/wallet/wallet/home/HomeFragment.kt

```diff
@@ -204,27 +207,46 @@ class HomeFragment : BasePageViewFragment(), SingleStateFragment<HomeState, Home
             }
           }
       }
-      else -> {}
+      else -> {
+        Column(
+          modifier = Modifier
+            .heightIn(0.dp, 480.dp)
+            .padding(horizontal = 16.dp)
+        ) {
+          Text(
+            text = stringResource(R.string.intro_transactions_header),
+            modifier = Modifier.padding(start = 8.dp, bottom = 16.dp, top = 26.dp),
+            style = MaterialTheme.typography.bodyMedium,
+            fontWeight = FontWeight.Bold,
+            color = WalletColors.styleguide_dark_grey
+          )
+          SkeletonLoadingTransactionCard()
+        }
+      }
     }
   }
 
   @Composable
   fun PromotionsList() {
-    if (!viewModel.activePromotions.isEmpty()) {
-      Text(
-        text = getString(R.string.intro_active_promotions_header),
-        fontSize = 14.sp,
-        fontWeight = FontWeight.Bold,
-        color = WalletColors.styleguide_dark_grey,
-        modifier = Modifier.padding(top = 27.dp, end = 13.dp, start = 24.dp)
-      )
-    }
+    Text(
+      text = getString(R.string.intro_active_promotions_header),
+      fontSize = 14.sp,
+      fontWeight = FontWeight.Bold,
+      color = WalletColors.styleguide_dark_grey,
+      modifier = Modifier.padding(top = 27.dp, end = 13.dp, start = 24.dp)
+    )
     LazyRow(
       contentPadding = PaddingValues(horizontal = 16.dp),
       horizontalArrangement = Arrangement.spacedBy(16.dp)
     ) {
-      items(viewModel.activePromotions) { promotion ->
-        PromotionsCardComposable(cardItem = promotion)
+      if (viewModel.activePromotions.isEmpty()) {
+        item {
+          SkeletonLoadingPromotionCard(hasVerticalList = true)
```

Comment:
The promotions list (not the skeleton) has a margin at the top. Might be good to include the same margin at the top so that the layout doesn't "jump around" when the skeletons disappear